scripts/linters/general_purpose_linter_test.py

```diff
@@ -409,3 +421,21 @@ def test_linter_ignore_pragma(self) -> None:
             [VALID_PY_IGNORE_PRAGMA_FILEPATH], FILE_CACHE)
         lint_task_report = linter.check_bad_patterns()
         self.assertFalse(lint_task_report.failed)
+
+    def test_check_bad_patterns_in_excluded_dirs(self) -> None:
+        bad_pattern_regexp = {
+            'regexp': re.compile(r'TODO[^\(]*[^\)][^:]*[^A-Z]+[^\w]*$'),
+            'message': 'Please link TODO comments to an issue '
+                    'in the format TODO(#issuenum): XXX. ',
+            'excluded_files': (),
+            'excluded_dirs': (LINTER_TESTS_DIR,)
+        }
+        check_status, error_message = (
```

Comment:
Why is "error_message" a list?

Reply:
It is actually a list of error messages. Renamed the variable now.